### DISCUSSION TITLE
Is anyone aware of the tags/attributes restrictions due to schema/device template use ?

### DISCUSSION THREAD

**Discussion by mmano** [apache/iotdb#12381] [Q&A]
Hi,
I'm quite sure that documentation doesn't explain anything about such a restriction. But if a schema/device template is applied onto a device node (v1.3 included), any tags/attributes can't be set on the timeseries related to this node. The whole issue is that we are not able to set the tags on the template itself. 
IMHO, giving up the templates vs. giving up the tags is a pain. No tags at all really weakens the data model expressiveness.
Do you have the knowledge about a gross roadmap ?

**Comment by q1255490** [comment]
Can't templates and tags co-exist? This one is really stupid!
